up42/auth.py

```diff
@@ -100,41 +123,55 @@ def __init__(
         except KeyError:
             self.authenticate = True
 
+        if all([arg in kwargs for arg in ["project_id", "project_api_key"]]) and auth_type == AuthType.PROJECT.value:
+            self.credentials_id = kwargs["project_id"]
+            self.credentials_key = kwargs["project_api_key"]
+            self.project_id = self.credentials_id
```

Comment:
self.kwargs.get("project_id")

Reply:
Please take a look to the alternative approach. Following your same logic but only set `self.project_id` if provided via arguments or cfg_file. Maybe, we need to add some further checking in the initialize as you suggest.